provider/pkg/gen/schema.go

```diff
@@ -522,25 +544,48 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 				fmt.Printf("Failed to build ID params for resource %s: %v\n", resourceTok, err)
 				return nil
 			}
-			resourceMeta.Read.Endpoint.Template = idPath
-			var idVals []resources.CloudAPIResourceParam
+			// TODO: This logic could be extracted into something more generic. Currently, the value computation
+			//       is always based on the GET method, so this may not be universally correct.
 			for _, k := range codegen.SortedKeys(vals) {
```

Comment:
I forget why we kept the values as a list in sorted order. Wonder if we should convert it to a map keyed off the name?

Reply:
Yeah, I think that would work also. There's definitely room to simplify this further.

Reply:
This is working for now, so let's follow up on that later